### DISCUSSION TITLE
How to assemble rip relative instruction?

### DISCUSSION THREAD

**Discussion by Symbai** [icedland/iced#268] [Q&A] [answered]
I want to assemble: `0x004040B7 - mov rax,[1202DB0h]` for example. Trying this: `c.mov(rax, __qword_ptr[0x1202DB0]);` works but the size of this instruction is 10 bytes, where it could have been 7 bytes instead: `48 8B 05 F2 EC DF 00` So how do I achieve the 7 bytes instruction using the assembler?

**Comment by wtfsck** [comment, marked as answer]
Not possible at the moment unless you use `db()` or use another register, eg. `RCX`.

I created an issue: https://github.com/icedland/iced/issues/269
